components/core/tests/test-kql.cpp

```diff
@@ -187,4 +187,44 @@ TEST_CASE("Test parsing KQL", "[KQL]") {
         auto failure = parse_kql_expression(incorrect_query);
         REQUIRE(nullptr == failure);
     }
+
+    SECTION("Escape sequences in column name") {
+        auto query = GENERATE(
+                "a\\.b.c: *",
+                "\"a\\.b.c\": *",
+                "a\\.b: {c: *}",
+                "\"a\\.b\": {\"c\": *}"
+        );
+        stringstream escaped_column_query{query};
+        auto filter
+                = std::dynamic_pointer_cast<FilterExpr>(parse_kql_expression(escaped_column_query));
+        REQUIRE(nullptr != filter);
+        REQUIRE(nullptr != filter->get_operand());
+        REQUIRE(nullptr != filter->get_column());
+        REQUIRE(false == filter->has_only_expression_operands());
+        REQUIRE(false == filter->is_inverted());
+        REQUIRE(FilterOperation::EQ == filter->get_operation());
+        REQUIRE(2 == filter->get_column()->get_descriptor_list().size());
+        auto it = filter->get_column()->descriptor_begin();
+        SPDLOG_INFO("{}", it->get_token());
+        REQUIRE(DescriptorToken{"a.b"} == *it++);
+        SPDLOG_INFO("{}", it->get_token());
+        REQUIRE(DescriptorToken{"c"} == *it++);
+    }
+
+    SECTION("Illegal escape sequences in column name") {
+        auto query = GENERATE(
+                //"a\\:*", this case is technically legal since ':' gets escaped
+                "\"a\\\":*",
+                "a\\ :*",
+                "\"a\\\" :*",
```

Comment:
Just confirming that the tokenizing done by ANTLR respects the space really. E.g. if the space were (unexpectedly) included in the column token then it would actually be legal, so we're confirming that this is not the case.